imap_processing/mag/l1a/mag_l1a.py

```diff
@@ -81,54 +148,120 @@ def process_packets(mag_l0_list: list[MagL0]) -> dict[str, list[MagL1a]]:
+        # TODO: Individual vectors should be sorted by day, not the whole packet
+        mago_is_primary = mag_l0.PRI_SENS == 0
+
+        mago_day = primary_day if mago_is_primary else secondary_day
+        magi_day = primary_day if not mago_is_primary else secondary_day
+
+        if mago_day not in mago:
+            mago[mago_day] = MagL1a(
                 True,
                 bool(mag_l0.MAGO_ACT),
-                primary_start_time,
-                mag_l0.PRI_VECSEC,
-                total_primary_vectors,
-                seconds_per_packet,
                 mag_l0.SHCOARSE,
-                primary_vectors,
+                primary_timestamped_vectors
+                if mago_is_primary
+                else secondary_timestamped_vectors,
+            )
+        else:
+            mago[mago_day].vectors = np.concatenate(
+                [
+                    mago[mago_day].vectors,
+                    (
+                        primary_timestamped_vectors
+                        if mago_is_primary
+                        else secondary_timestamped_vectors
+                    ),
+                ]
             )
 
-            magi_l1a = MagL1a(
+        if magi_day not in magi:
+            magi[magi_day] = MagL1a(
                 False,
                 bool(mag_l0.MAGI_ACT),
-                secondary_start_time,
-                mag_l0.SEC_VECSEC,
-                total_secondary_vectors,
-                seconds_per_packet,
                 mag_l0.SHCOARSE,
-                secondary_vectors,
+                primary_timestamped_vectors
+                if not mago_is_primary
+                else secondary_timestamped_vectors,
             )
-        # Primary sensor is MAGI
         else:
-            magi_l1a = MagL1a(
-                False,
-                bool(mag_l0.MAGI_ACT),
-                primary_start_time,
-                mag_l0.PRI_VECSEC,
-                total_primary_vectors,
-                seconds_per_packet,
-                mag_l0.SHCOARSE,
-                primary_vectors,
+            magi[magi_day].vectors = np.concatenate(
+                [
+                    magi[magi_day].vectors,
+                    (
+                        primary_timestamped_vectors
+                        if not mago_is_primary
+                        else secondary_timestamped_vectors
+                    ),
+                ]
             )
 
-            mago_l1a = MagL1a(
-                True,
-                bool(mag_l0.MAGO_ACT),
-                secondary_start_time,
-                mag_l0.SEC_VECSEC,
-                total_secondary_vectors,
-                seconds_per_packet,
-                mag_l0.SHCOARSE,
-                secondary_vectors,
-            )
+    return {"mago": mago, "magi": magi}
 
-        mago.append(mago_l1a)
-        magi.append(magi_l1a)
 
-    return {"mago": mago, "magi": magi}
+def generate_dataset(mag_l1a: MagL1a, dataset_attrs: dict):
```

Comment:
For a second, I thought there is a circular loop happening here because of variable and function named same thing here. I thought `mag_l1a` was calling function `mag_l1a` which would result in circular loop but you were not. can you rename this variable name to something else to avoid this confusion?

Reply:
Can you add a test function where it calls `mag_l1a()`? It may help in understanding which order these functions are run.

Reply:
Yes, I'll also rename and add comments to the functions. I agree that it's confusing, I tried to be clear but there are a lot of `mag_l1a`s running around still!

Reply:
I appreciate that!

Reply:
```suggestion
def generate_dataset(mag_l1a: MagL1a, dataset_attrs: dict) -> xr.Dataset:
```